bin/docker.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+# Author: Valentin Kuznetsov <vkuznet@gmail.com>
+# helper script to handle docker actions
+
+if [ $# -ne 4 ]; then
+     echo "Usage: docker.sh <action> <service> <tag> <registry> "
+     echo "Supported actions: build, build-stable, push"
+     exit 1;
+fi
+
+# helper function to match the pattern X.Y.Z where X, Y, and Z are numbers
+function match_tag {
+  local tag=$1
+  if [[ $tag =~ ^([0-9]+)\.([0-9]+)\.([0-9]+)$ ]]; then
+    return 0
+  elif [[ $tag =~ ^([0-9]+)\.([0-9]+)\.([0-9]+)\.([0-9]+)$ ]]; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+# make input variable assignments
+action=$1
+service=$2
+tag=$3
+registry=$4
+rurl=$registry/cmsweb/$service
+
+# check if action is supported
+case "$action" in
+  "build"|"build-stable"|"push"|"remove"|"delete")
+    ;;
+  *)
+    echo "action: '$action' is not supported"
+    exit 1
+    ;;
+esac
+
+# determine suffix of image tag
+suffix=""
+if match_tag "$tag"; then
+    suffix="-stable"
+fi
+
+# check build/push actions and build the image(s) if it will be required
+if [ "$action" == "build" ]; then
+    echo "check if image with service=$service and tag=$tag exist ..."
+    image_exist=`docker images | grep $tag | grep $service`
+    if [ -z "$image_exist" ]; then
+        echo "action: docker build --build-arg TAG=${tag} --tag ${rurl}:${tag}"
+        docker build --build-arg TAG=${tag} --tag ${rurl}:${tag} .
+    fi
+    docker images | grep $tag | grep $service
+fi
+if [ "$action" == "build-stable" ] || ( [ "$action" == "build" ] && [ -n "$suffix" ] ); then
+    if [ -z "$suffix" ]; then
+        echo "Building stable image for tag=$tag is not appropriate as tag is not matched X.Y.Z or X.Y.Z.P pattern"
+        exit 0
+    fi
+    echo "check if image with service=$service and tag=${tag}${suffix} exist ..."
+    image_exist=`docker images | grep ${tag}${suffix} | grep $service`
+    if [ -z "$image_exist" ]; then
+        echo "action: docker build --build-arg TAG=${tag} --tag ${rurl}:${tag}${suffix}"
+        docker build --build-arg TAG=${tag} --tag ${rurl}:${tag}${suffix} .
+    fi
+    docker images | grep ${tag}${suffix} | grep $service
+fi
+if [ "$action" == "push" ] || ( [ "$action" == "push" ] && [ -n "$suffix" ] ); then
+    image_exist=`docker images | grep $tag | grep $service`
+    if [ -z "$image_exist" ]; then
+        echo "Images ${rurl}:${tag}${suffix} and/or ${rurl}:${tag}${suffix} are not found"
+        exit 0
+    fi
+    if [ -n "$image_exist" ]; then
+        echo "action: docker push ${rurl}:${tag}"
```

Comment:
TODO: After testing everything and before merging this PR, we need to remember to remove this "echo" statement.